src/mappings/shared/handleTokenEntity.ts

```diff
@@ -0,0 +1,52 @@
+import { create, getOptional } from '@kodadot1/metasquid/entity'
+import md5 from 'md5'
+import { CollectionEntity as CE, NFTEntity as NE, TokenEntity as TE } from '../../model'
+import { warn } from '../utils/logger'
+import { Context } from '../utils/types'
+
+const OPERATION = 'TokenEntity' as any
+
+export async function handleTokenEntity(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
+  const nftMedia =
+    nft.image ??
+    nft.media ??
+    nft.meta?.image ??
+    nft.meta?.animationUrl ??
+    nft.resources[0].src ??
+    nft.resources[0].thumb
+  if (!nftMedia || nftMedia === '') {
+    warn(OPERATION, `MISSING NFT MEDIA ${nft.id}`)
+    await new Promise((resolve) => setTimeout(resolve, 5000))
```

Comment:
Why are you waiting 5 sec here?

Reply:
oops sorry. it was for debugging, i wanted to make the log slow down.
of course it doesn't belong in production

Reply:
removed